File: csharp/multi_threaded.cs
using System;
using System.Diagnostics;
using System.Threading;

class Program
{
    static void ConvertMultiThreaded()
    {
        Stopwatch stopwatch = Stopwatch.StartNew();

        // Criar threads para simular conversão simultânea
        int numberOfThreads = 5;  // Exemplo com 5 threads
        Thread[] threads = new Thread[numberOfThreads];
        for (int i = 0; i < numberOfThreads; i++)
        {
            threads[i] = new Thread({
                // !Lógica de conversão de documentos aqui
            });
            threads[i].Start();
        }

        // Aguardar a conclusão de todas as threads
        foreach (Thread thread in threads)
        {
            thread.Join();
        }

        stopwatch.Stop();
        Console.WriteLine($"Tempo de execução (multi-threaded): {stopwatch.Elapsed.TotalSeconds} segundos");
    }

    static void Main()
    {
        ConvertMultiThreaded();
    }
}

File: c++/multi_threaded.cpp
#include <iostream>
#include <vector>
#include <chrono>
#include <thread>

void convertMultiThreaded() {
    auto startTime = std::chrono::high_resolution_clock::now();

    // Criar threads para simular conversão simultânea
    int numberOfThreads = 5;  // Exemplo com 5 threads
    std::vector<std::thread> threads;
    for (int i = 0; i < numberOfThreads; ++i) {
        threads.emplace_back([]() {
            // !Lógica de conversão de documentos aqui
        });
    }

    // Aguardar a conclusão de todas as threads
    for (std::thread& thread : threads) {
        thread.join();
    }

    auto endTime = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = endTime - startTime;
    std::cout << "Tempo de execução (multi-threaded): " << duration.count() << " segundos" << std::endl;
}

int main() {
    convertMultiThreaded();
    return 0;
}